server/ws_tunnel.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"errors"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/elazarl/goproxy"
+)
+
+/*
+ * The purpose of this feature is  to allow tunneling of arbitrary TCP connections using the wstunnel protocol.
+ * One use case for this is to allow tunneling of all Pelion edge traffic so through a restrictive corporate firewall.
+ */
+
+type WSTunnelConfig struct {
+	ListenAddr string // Address of the listening port
+	TunnelAddr string // Address of the remote wsTunnel server
+}
+
+// StartWSTunnel starts a server that accepts HTTP CONNECT method to proxy arbitrary TCP connections.
+//
+func StartWSTunnel(config *WSTunnelConfig) error {
+	proxy := goproxy.NewProxyHttpServer()
+
+	_, err := url.Parse(config.TunnelAddr)
```

Comment:
`config.TunnelAddr` seems unnecessary here. And this looks more like a client configuration parameter instead of a server configuration parameter.